stock_valuation_fifo_lot/tests/test_stock_valuation_fifo_lot.py

```diff
@@ -0,0 +1,474 @@
+        return_picking_wizard_form = Form(
+            self.env["stock.return.picking"].with_context(
+                active_ids=original_picking.ids,
+                active_id=original_picking.id,
+                active_model="stock.picking",
+            )
+        )
+        return_picking_wizard = return_picking_wizard_form.save()
+        return_picking_wizard.product_return_moves.write({"quantity": return_qty})
+        return_picking_wizard_action = return_picking_wizard.create_returns()
+        return_picking = self.env["stock.picking"].browse(
+            return_picking_wizard_action["res_id"]
+        )
+        return_move = return_picking.move_ids
+        return_move.move_line_ids.qty_done = return_qty
+        return_picking.button_validate()
+        return return_move
+
+    def test_stock_valuation_fifo_lot(self):
+        _, move_in = self.create_picking(
+            self.supplier_location,
+            self.stock_location,
+            self.picking_type_in,
+            ["001"],
+            100.0,
+        )
+        self.assertEqual(
+            move_in.stock_valuation_layer_ids.value,
+            500.0,
+            "Stock valuation for the first receipt should be 500.0",
+        )
+
+        _, move_in = self.create_picking(
+            self.supplier_location,
+            self.stock_location,
+            self.picking_type_in,
+            ["002"],
+            200.0,
+        )
+        self.assertEqual(
+            move_in.stock_valuation_layer_ids.value,
+            1000.0,
+            "Stock valuation for the second receipt should be 1000.0",
+        )
+        self.assertEqual(
+            self.product.standard_price,
+            100.0,
+            "Standard price should be set to 100.0 after first receipt",
+        )
+
+        # Create delivery for lot 002
+        _, move_out = self.create_picking(
+            self.stock_location,
+            self.customer_location,
+            self.picking_type_out,
+            ["002"],
+            is_receipt=False,
+        )
+        self.assertEqual(
+            abs(move_out.stock_valuation_layer_ids.value),
+            1000.0,
+            "Stock valuation for the delivery should be 1000.0",
+        )
+
+        # Test return receipt
+        receipt_picking_3, move = self.create_picking(
+            self.supplier_location,
+            self.stock_location,
+            self.picking_type_in,
+            ["003"],
+            300.0,
+        )
+        self.assertEqual(
+            move.stock_valuation_layer_ids.value,
+            1500.0,
+            "Stock valuation for the third receipt should be 1500.0",
+        )
+
+        return_move = self.transfer_return(receipt_picking_3, 5.0)
+        self.assertEqual(
+            abs(return_move.stock_valuation_layer_ids.value),
+            1500.0,
+            "Stock valuation after return should be 1500.0",
+        )
+
+    def test_receive_deliver_return_lot(self):
```

Comment:
I didn't quite get the point of this test. Does it prove anything when all lots have the same cost?

Reply:
This was for the case like we receive serials `001`, `002` and `003` for an incoming move.Then, Deliver `002`, return `002` and deliver `002` again. In this case, we want to check the remaining_qty was taken from return move's SVL instead of first incoming move's SVL because there remaining qty of first incoming move's SVL is only for both `001` and `002` serials now.